### PR TITLE
Introduce metababel.h

### PR DESCRIPTION
**Pull request by TApplencourt**

### CODE REVIEW

File: bin/metababel [hunk 1]
```diff
@@ -4,6 +4,7 @@ require 'yaml'
 require 'optparse'
 require 'erb'
 require 'metababel'
+require 'fileutils'
 
 def validate_model(yaml)
   raise "Invalide Model #{yaml}" unless yaml[:stream_classes].all? do |d|
```
Comment (DonAurelio) [inline bin/metababel:10]: Invalid ?

Comment (TApplencourt) [reply bin/metababel:10]: Yes! Can you `add a suggestion`? :)  (https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request point 6, )

File: template/downstream.c.erb
Comment (DonAurelio) [inline template/downstream.c.erb:1]: I am wondering, why the downstream.c do not import downtream.h?

Comment (TApplencourt) [reply template/downstream.c.erb:1]: Good catch. We should indeed. Can you add a code change suggestion? :D

File: template/upstream.c.erb [hunk 2]
```diff
@@ -1,5 +1,5 @@
-#include "component.h"
-#include "upstream.h"
+#include  <metababel/btx_component.h>
```
Comment (DonAurelio) [inline template/upstream.c.erb:1]: Extra space

**Review by DonAurelio — APPROVED** [review]

File: bin/metababel (hunk 1 above)
Comment (DonAurelio) [inline bin/metababel:10]:
```suggestion
  raise "Invalid Model #{yaml}" unless yaml[:stream_classes].all? do |d|
```

File: template/upstream.c.erb (hunk 2 above)
Comment (DonAurelio) [inline template/upstream.c.erb:1]:
```suggestion
#include <metababel/btx_component.h>
```

File: template/downstream.c.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#include "component.h"
 #include "utlist.h"
 #include <babeltrace2/babeltrace.h>
+#include <metababel/btx_component.h>
```
Comment (TApplencourt) [inline template/downstream.c.erb:3]:
```suggestion
#include <metababel/btx_component.h>
#include <metababel/btx_downstream.h>
```

**Pull request merged**